llvm/lib/Target/AMDGPU/GCNRewritePartialRegUses.cpp

```diff
@@ -436,14 +426,6 @@ bool GCNRewritePartialRegUsesImpl::rewriteReg(Register Reg) const {
     if (Inserted)
       SubRegRC = TRI->getSubRegisterClass(RC, SubReg);
 
-    if (SubRegRC) {
-      if (const TargetRegisterClass *OpDescRC = getOperandRegClass(MO)) {
```

Comment:
I can't find any test case where this is needed. If getSubRegisterClass returns an exact match (i.e. not a superclass of what you ask for) then I guess it doesn't need to be constrained any more than the original RC was.